src/include/planner/insert_plan.h

```diff
@@ -29,27 +30,34 @@ class InsertStatement;
 namespace planner {
 class InsertPlan : public AbstractPlan {
  public:
-  // This constructor takes in neither a project info nor a tuple
-  // It must be used when the input is a logical tile
-  explicit InsertPlan(storage::DataTable *table, oid_t bulk_insert_count = 1);
-
-  // This constructor takes in a project info
-  explicit InsertPlan(
-      storage::DataTable *table,
-      std::unique_ptr<const planner::ProjectInfo> &&project_info,
-      oid_t bulk_insert_count = 1);
-
-  // This constructor takes in a tuple
-  explicit InsertPlan(storage::DataTable *table,
-                      std::unique_ptr<storage::Tuple> &&tuple,
-                      oid_t bulk_insert_count = 1);
-
-  explicit InsertPlan(
-      storage::DataTable *table, const std::vector<std::string> *columns,
-      const std::vector<std::vector<std::unique_ptr<expression::AbstractExpression>>> *
-          insert_values);
-
-  // Get a varlen pool (will construct the pool only if needed)
+  // Construct when input is a logical tile
+  InsertPlan(storage::DataTable *table, oid_t bulk_insert_count = 1)
+    : target_table_(table), bulk_insert_count_(bulk_insert_count) {}
+
+  // Construct with a project info
+  InsertPlan(storage::DataTable *table,
```

Comment:
I am not sure whether moving the implementation of some constructors to the header file is a good idea. Any specific reason for doing this?

Reply:
These constructors that I moved are doing nothing much other than setting member variables, so I moved them to headers.  Nothing functional. Do we have a rule on this? I could apply accordingly. Thanks for the comment. I did not think much about it when I changed. 👍

Reply:
I don't think there is a hard rule. Just in general, it would be nice to keep it consistent and make the header shorter.